src/timesync/fuzz-nts-extfields.c

```diff
@@ -0,0 +1,80 @@
+#include <string.h>
+#include <stdio.h>
+#include <assert.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+#include "fuzz.h"
+
+#include "nts.h"
+#include "nts_crypto.h"
+#include "nts_extfields.h"
+
+static void eat(volatile const uint8_t* buf, size_t size) {
+        if (buf) while (size) (void)buf[size--];
+}
+
+/* this program does no sanity checking as it is meant for fuzzing only */
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+        uint8_t buffer[1280];
+        int len = size > sizeof(buffer)? sizeof(buffer) : size;
+
+        memcpy(buffer, data, len);
+        if (len < 48) return 0;
+
+        struct NTS_Query nts = (struct NTS_Query) {
+                .cipher = *NTS_get_param(NTS_AEAD_AES_SIV_CMAC_256),
+                .c2s_key = (void*)"01234567890abcdef",
+                .s2c_key = (void*)"01234567890abcdef",
+        };
+
+        /* fuzz the NTS extension field parser */
+        struct NTS_Receipt rcpt = { 0, };
+        if (NTS_parse_extension_fields(&buffer, len, &nts, &rcpt)) {
+                for (int i = 0; i < 8; i++)
+                        eat(rcpt.new_cookie[i].data, rcpt.new_cookie[i].length);
+                eat(*rcpt.identifier, 32);
+        }
+
+        return 0;
+}
+
+/* null cipher */
+
+#define BLKSIZ 16
+
+int NTS_encrypt(uint8_t *ctxt,
```

Comment:
the code below assumes that ctxt is 16 byte long hence write this as `uint8_t ctxt[static BLKSIZ]`